independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/AroundInvokeInvocationContext.java

```diff
@@ -65,4 +80,71 @@ public Object proceed() throws Exception {
         }
     }
 
+    class NextAroundInvokeInvocationContext implements ArcInvocationContext {
+
+        private final int position;
+        protected Object[] parameters;
+
+        public NextAroundInvokeInvocationContext(int position) {
+            this.position = position;
+        }
+
+        @Override
+        public Object proceed() throws Exception {
+            return AroundInvokeInvocationContext.this.proceed(position);
+        }
+
+        @Override
+        public Object getTarget() {
+            return AroundInvokeInvocationContext.this.getTarget();
+        }
+
+        @Override
+        public Object getTimer() {
+            return AroundInvokeInvocationContext.this.getTimer();
+        }
+
+        @Override
+        public Method getMethod() {
+            return AroundInvokeInvocationContext.this.getMethod();
+        }
+
+        @Override
+        public Constructor<?> getConstructor() {
+            return AroundInvokeInvocationContext.this.getConstructor();
+        }
+
+        @Override
+        public Object[] getParameters() {
+            return parameters;
+        }
+
+        @Override
+        public void setParameters(Object[] params) {
+            validateParameters(metadata.method, params);
+            this.parameters = params;
```

Comment:
Shouldn't this also delegate to `AroundInvokeInvocationContext.this`?

This way, interceptors that are not last in the chain will effectively not be allowed to modify the arguments I think? Or am I missing something?

Reply:
That's a good point. I just kept the previous behavior. But it was probably wrong. I think that we need a better test... (until we run the TCK ;-)

Reply:
Ah we already did this? Sorry, didn't notice that. I'll need to look at the commit locally to better understand.